feat: built about-me screen

iDev-20 opened the pull request:
This PR introduces the About Me screen where users can:
Enter their first name
Enter their other names
Provide a short self-description
Submit the form via a button

@being-eyram @Dev-Tyron for review

<img width="1080" height="2400" alt="about_me_screen_screenshot" src="https://github.com/user-attachments/assets/148e953a-b244-4d0b-93d7-08dcb0863102" />
<img width="1080" height="2400" alt="filled_about_me_screen_screenshot" src="https://github.com/user-attachments/assets/e034634a-6494-4401-b098-692f75204dad" />

being-eyram commented on about_me_screen_screenshot.png:
The font used for  "About Me" title is different from the one specified in the design. 
Kindly check that.

being-eyram commented on lib/views/onboarding/about_me_screen.dart:
```diff
@@ -0,0 +1,112 @@
+                    keyboardType: TextInputType.name,
+                    textInputAction: TextInputAction.next,
+                    onChanged: (c) {
+                      setState(() {});
```
Prefer TextEditingController.addListener to setState

iDev-20 commented:
I've made the following updates based on the feedback:

Replaced setState with TextEditingController.addListener in the About Me screen.
Updated the font for the "About Me" title to match the design spec.
Created a reusable AppPage widget and applied it in the About Me screen.
Handled async navigation safely in the splash screen using if (!mounted) return.

Please let me know if there's anything else to tweak. Thanks! 🙌
